### PR TITLE
T1783

### PR DESCRIPTION
**Pull request by secretrobotron**

### CODE REVIEW

File: src/editor/base-editor.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
+/* This Source Code Form is subject to the terms of the MIT license
```
Comment (secretrobotron) [inline src/editor/base-editor.js:1]: Note: this was just clipped out of `src/editor/editor.js` to put it in its own file for cleanliness.

File: src/editor/editor.js [hunk 2]
```diff
@@ -5,363 +5,20 @@
...
-                          KeysUtils.ESCAPE
-                        ];
+  var __editors = {};
 
   /**
    * Namespace: Editor
    */
   var Editor = {
 
-    /**
```
Comment (secretrobotron) [inline src/editor/editor.js:30]: Moved to `src/editor/base-editor.js`.

File: src/editor/editor.js [hunk 3]
```diff
@@ -5,363 +5,20 @@
...
-
-        // If a close event existed on the events object passed into the constructor, call it
-        if ( events.close ) {
-          events.close.apply( extendObject, arguments );
-        }
-
-        extendObject.dispatch( "closed" );
-      };
-
-      extendObject.defaultLayouts = __defaultLayouts.cloneNode( true );
```
Comment (secretrobotron) [inline src/editor/editor.js:85]: Moved to `/src/editor/trackevent-editor.js`.

File: src/editor/trackevent-editor.js [hunk 4]
```diff
@@ -0,0 +1,316 @@
+/* This Source Code Form is subject to the terms of the MIT license
```
Comment (secretrobotron) [inline src/editor/trackevent-editor.js:1]: Note: this was just clipped out of `src/editor/editor.js` to put it in its own file for cleanliness and extendability.

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @humphd received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/72ae9916ca16d50/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_try** from @humphd received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/0716effa2ebe133/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/72ae9916ca16d50/output.txt

Total script time: 0.30 mins

File: src/editor/default.js [hunk 5]
```diff
@@ -68,10 +68,10 @@ define( [ "text!./default.html", "editor/editor" ],
               }
               else {
                 if ( element.type === "checkbox" ) {
-                  _this.attachCheckboxChangeHandler( element, trackEvent, name, updateTrackEventWithoutTryCatch );
+                  _this.attachCheckboxChangeHandler( element, trackEvent, name, updateTrackEventWithTryCatch );
                 }
                 else {
-                  _this.attachInputChangeHandler( element, trackEvent, name, updateTrackEventWithoutTryCatch );
+                  _this.attachInputChangeHandler( element, trackEvent, name, updateTrackEventWithTryCatch );
```
Comment (humphd) [inline src/editor/default.js]: If you're removing the _Without_ code path, why keep that naming convention?  That is, why not just call it updateTrackEvent?

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/0716effa2ebe133/output.txt

Total script time: 1.68 mins

Your job has been submitted to the swarm! see http://norway.proximity.on.ca/user/butter

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @humphd received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/9abfc6b16899b2c/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/9abfc6b16899b2c/output.txt

Total script time: 1.01 mins

Server is running on http://pr657.butyr.org/templates/test.html

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_try** from @cadecairos received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/c805d9e77971314/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/c805d9e77971314/output.txt

Total script time: 1.98 mins

Your job has been submitted to the swarm! see http://norway.proximity.on.ca/user/butter

**Pull request closed without merging**